### PR TITLE
English translation

### PR DESCRIPTION
**Pull request by skade**
This translates everything not yet translated from german to english.

Greetings from Mexico.

### CODE REVIEW

**Comment by skade** [comment]
cc @ericho

**Comment by ericho** [comment]
Awesome!! I'll start with the translation to spanish 😄

File: presentation/chapters/en-US/drop-panic-abort.chapter [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+In case of a panic, the following happens:
+
+\* The current thread immediately halts
+\* The stack is unwound
+\* All affected values are dropped and their destructors run
+
+---
+
+Panics are implementation-wise similar to C++-Exceptions, but should only be used for fatal errors. They cannot be (normally) cought.
```
Comment (Hoverbear) [inline presentation/chapters/en-US/drop-panic-abort.chapter]: caught

File: presentation/chapters/en-US/dynamic-dispatch.chapter [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+-   Object-safe traits are *not* allowed to require `Self: Sized`
+-   All methods are object-safe
+    -   They have no type parameters
+    -   They don't use `Self`
+    
+---
+
+## Futher properties
+
+-   as trait objects know about their exact type at runtime, they support downcasts.
```
Comment (Hoverbear) [inline presentation/chapters/en-US/dynamic-dispatch.chapter]: "As"

File: presentation/chapters/en-US/effective-rust.chapter [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+---
+
+In some cases, `Drain` can be used, an iterator that drains values one-by-one from a collection and shrinks it at the same time.
+
+---
+
+## Compile times
+
+`rustc` is not the fastest compiler, much time is spent in LLVM with code generation and optiomisation.
```
Comment (Hoverbear) [inline presentation/chapters/en-US/effective-rust.chapter]: optimisation

File: presentation/chapters/en-US/embedded.chapter [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+Copper: Programming Microcontrollers with Rust and Cortex-M.
+
+<https://japaric.github.io/copper/>
+
+---
+
+## Projects
+
+Tock, an operating system for Cortex-M-Prozessoren: [tockos.org](tockos.org)
```
Comment (Hoverbear) [inline presentation/chapters/en-US/embedded.chapter]: Cortex-M Processors

File: presentation/chapters/en-US/embedded.chapter [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+---
+
+## Problems
+
+-   LLVM-Target-Support
```
Comment (Hoverbear) [inline presentation/chapters/en-US/embedded.chapter]: LLVM-Target Support

File: presentation/chapters/en-US/embedded.chapter [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    -   LLVM is generally good on targets that Apple uses
+    -   This doesn't mean that LLVM is in any way hostile!
+-   Unsolved requirements in Rust, especially bitfields
+-   Inline-Assembly is unstable
```
Comment (Hoverbear) [inline presentation/chapters/en-US/embedded.chapter]: Inline Assembly

File: presentation/chapters/en-US/embedded.chapter [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+-   Core-Developers are Desktop- and Serverdevelopers and don't always have enough experience
```
Comment (Hoverbear) [inline presentation/chapters/en-US/embedded.chapter]: Core Developers are Desktop and Server developers

File: presentation/chapters/en-US/embedded.chapter [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    -   "No solution is better then a bad solution!"
+-   We need Vendor-buy-in
```
Comment (Hoverbear) [inline presentation/chapters/en-US/embedded.chapter]: vendor buy-in

File: presentation/chapters/en-US/embedded.chapter [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+---
+
+## Solution
+
+-   Rust is a driving Force behind better Target-Support in LLVM
+    -   Multiple Targets, espcially AVR, to be released in the next versions
+-   Enthusiastic Embedded-Community
```
Comment (Hoverbear) [inline presentation/chapters/en-US/embedded.chapter]: embedded community

File: presentation/chapters/en-US/ffi.chapter [hunk 10]
```diff
@@ -0,0 +1,218 @@
...
+
+## System-libc
+
+<pre><code data-source="chapters/shared/code/ffi/2.rs" data-trim="hljs rust"></code></pre>
+
+---
+
+## Opaque types
+
+We use empty Enums here. The cannot be constructed or read: they have no variants!
```
Comment (Hoverbear) [inline presentation/chapters/en-US/ffi.chapter]: "The" -> "They"

File: presentation/chapters/en-US/ffi.chapter [hunk 11]
```diff
@@ -0,0 +1,218 @@
...
+
+---
+
+## Callbacks
+
+<pre><code data-source="chapters/shared/code/ffi/6.rs" data-trim="hljs rust"></code></pre>
+
+---
+
+## Utility-Functions
```
Comment (Hoverbear) [inline presentation/chapters/en-US/ffi.chapter]: Utility Functions

File: presentation/chapters/en-US/ffi.chapter [hunk 12]
```diff
@@ -0,0 +1,218 @@
...
+
+## Why like this?
+
+-   Structs with one field vanish at runtime
+-   Isolates lifecycle management of the pointer from the rest of the code
+-   Older Rust versions modify the struct layout if `Drop` is implemented
+
+---
+
+## Handling of C-Style-Errors
```
Comment (Hoverbear) [inline presentation/chapters/en-US/ffi.chapter]: "C Style Errors"

File: presentation/chapters/en-US/ffi.chapter [hunk 13]
```diff
@@ -0,0 +1,218 @@
...
+
+> Error handling styles in C can be organized into one of several distinct styles, such as popular or correct. Some examples of each.
+
+[Ted Unangst](http://www.tedunangst.com/flak/post/to-errno-or-to-error)
+
+---
+
+## Handling of leveldb-Errors
```
Comment (Hoverbear) [inline presentation/chapters/en-US/ffi.chapter]: "Leveldb Errors"

File: presentation/chapters/en-US/ffi.chapter [hunk 14]
```diff
@@ -0,0 +1,218 @@
...
+
+## Problems
+
+-   `state: *c_void` is a way of expressing polymorphism in C
+-   How do I ensure that, `c_void` always refers to the same type?
+-   How do I implement the polymorphism?
+
+---
+
+## Generics, Traits and raw Boxen
```
Comment (Hoverbear) [inline presentation/chapters/en-US/ffi.chapter]: "Boxen" is probably not the right term. "Raw Boxes"?

File: presentation/chapters/en-US/imports-modules-and-visibility.chapter [hunk 15]
```diff
@@ -1 +1,140 @@
...
+
+## Module-Imports
+
+It is possible to import the module instead and qualify every use.
+
+<pre><code data-source="chapters/shared/code/imports-modules-and-visibility/2.rs" data-trim="hljs rust"></code></pre>
+
+---
+
+## Glob-Imports
```
Comment (Hoverbear) [inline presentation/chapters/en-US/imports-modules-and-visibility.chapter]: "Glob Imports"

File: presentation/chapters/en-US/imports-modules-and-visibility.chapter [hunk 16]
```diff
@@ -1 +1,140 @@
...
+
+---
+
+## Example
+
+<pre><code data-source="chapters/shared/code/imports-modules-and-visibility/7.rs" data-trim="hljs rust"></code></pre>
+
+---
+
+## Moving the Module to a Seperate File
```
Comment (Hoverbear) [inline presentation/chapters/en-US/imports-modules-and-visibility.chapter]: Separate*

File: presentation/chapters/en-US/lifetimes.chapter [hunk 17]
```diff
@@ -1 +1,121 @@
...
+---
+
+That is not necessary.
+
+---
+
+## Lifetimes
+
+-   Lifetimes describe the time that values remain in memory
+-   They describe - they cannot force or change anythin
```
Comment (Hoverbear) [inline presentation/chapters/en-US/lifetimes.chapter]: anything

File: presentation/chapters/en-US/lifetimes.chapter [hunk 18]
```diff
@@ -1 +1,121 @@
...
+The correct struct definition is:
+
+<pre><code data-source="chapters/shared/code/lifetimes/2.rs" data-trim="hljs rust"></code></pre>
+
+---
+
+Wrapper is now:
+
+* Generic over the type T
+* as well as a lifetime 'a (its own)
```
Comment (Hoverbear) [inline presentation/chapters/en-US/lifetimes.chapter]: Capitalize these list items. :) 


* "As"
* "The"

File: presentation/chapters/en-US/match.chapter [hunk 19]
```diff
@@ -1,30 +1,51 @@
-# Control Flow With Match
+# Control-Flow with `match`
```
Comment (Hoverbear) [inline presentation/chapters/en-US/match.chapter]: "Control Flow with `match`"

File: presentation/chapters/en-US/match.chapter [hunk 20]
```diff
@@ -1,30 +1,51 @@
...
-Solution: Match or pass along.
+
+---
+
+Results carry a special marker: they must not be ignored!
+
+
+<pre><code data-source="chapters/shared/code/match/5.rs" data-trim="hljs rust"></code></pre>
+
+Solution: match or pass on.
```
Comment (Hoverbear) [inline presentation/chapters/en-US/match.chapter]: What about `.ok()` which demotes them to an `Option<T>`?

File: presentation/chapters/en-US/overview.chapter [hunk 21]
```diff
@@ -1 +1,131 @@
...
+-   Adopted by Mozilla
+-   Presented to the general public as version 0.4 in 2012
+-   Looked a bit Go-like back then
+
+---
+
+## Focus
+
+-   Rust has lost many features from 2012 to 2014
+-   garbage collector, evented runtime, complex error handling
```
Comment (Hoverbear) [inline presentation/chapters/en-US/overview.chapter]: These list points need capitals.

File: presentation/chapters/en-US/overview.chapter [hunk 22]
```diff
@@ -1 +1,131 @@
...
+
+## Development
+
+-   Always together with a larger project (e.g. Servo)
+-   Early adoption of regular releases, deprecations and an RFC process
+-   Good software development methodologies
+
+---
+
+## Release-Mode
```
Comment (Hoverbear) [inline presentation/chapters/en-US/overview.chapter]: "Release Method"?

File: presentation/chapters/en-US/overview.chapter [hunk 23]
```diff
@@ -1 +1,131 @@
...
+-   After 6 weeks of testing, beta becomes stable
+-   Guaranteed backwards-compatibility
+-   Makes small iterations easier
+
+---
+
+## Goals
+
+-   Explicit over implicit
+-   predictable runtime behaviour
```
Comment (Hoverbear) [inline presentation/chapters/en-US/overview.chapter]: Capitalize these please :)

File: presentation/chapters/en-US/overview.chapter [hunk 24]
```diff
@@ -1 +1,131 @@
...
+-   No illegal memory access
+-   Deallocation is automated
+-   Warning: memory leaks are *safe* by that definition!
+
+---
+
+## Concurrent
+
+-   "Concurrency without fear"
+-   the type system detects concurrent access to data and requires synchronisation
```
Comment (Hoverbear) [inline presentation/chapters/en-US/overview.chapter]: Capitalization

File: presentation/chapters/en-US/overview.chapter [hunk 25]
```diff
@@ -1 +1,131 @@
...
+-   also: Rust detects when unsynchronised access is safely possible!
+-   protection from data races!
+
+---
+
+## Fast
+
+-   these properties are guaranteed at compile time and have no runtime cost!
```
Comment (Hoverbear) [inline presentation/chapters/en-US/overview.chapter]: Capitalization

File: presentation/chapters/en-US/overview.chapter [hunk 26]
```diff
@@ -1 +1,131 @@
...
+-   optimizing compiler based on LLVM
+-   features with runtime cost are explicit and hard to activate "by accident"
+-   no reflection (Dracula-Language)
+-   static dispatch per default
+
+---
+
+## Pragmatic
+
+-   sublanguage for unsafe memory access
```
Comment (Hoverbear) [inline presentation/chapters/en-US/overview.chapter]: Capitalization

File: presentation/chapters/en-US/ownership.chapter [hunk 27]
```diff
@@ -1 +1,153 @@
...
+
+---
+
+## Working with moves: copy instead of move
+
+<pre><code data-source="chapters/shared/code/ownership/4.rs" data-trim="hljs rust"></code></pre>
+
+---
+
+Copy is meant for data that can be quickly copied in memory (using memcopy) and are allowed to be copied (e.g.: not Filepointers).
```
Comment (Hoverbear) [inline presentation/chapters/en-US/ownership.chapter]: File pointers

File: presentation/chapters/en-US/stack-and-heap.chapter [hunk 28]
```diff
@@ -0,0 +1,39 @@
+# Stack and Heap
+
+---
+
+Rust defaults to allocation on the stack
+
+---
+
+## Stack-Allocation
```
Comment (Hoverbear) [inline presentation/chapters/en-US/stack-and-heap.chapter]: Stack Allocation

File: presentation/chapters/en-US/stack-and-heap.chapter [hunk 29]
```diff
@@ -0,0 +1,39 @@
...
+
+<pre><code data-source="chapters/shared/code/stack-and-heap/1.rs" data-trim="hljs rust"></code></pre>
+
+---
+
+## Box
+
+Heap-Allocation is represented by the type `Box`.
```
Comment (Hoverbear) [inline presentation/chapters/en-US/stack-and-heap.chapter]: Heap Allocation

File: presentation/chapters/en-US/unsafe.chapter [hunk 30]
```diff
@@ -0,0 +1,65 @@
...
+## Traps of `unsafe`
+
+-   Not all examples are that simple. `unsafe` *must* guarantee the invariants that Rust expects.
+-   This *especially* applies to ownership and mutable borrowing
+-   `unsafe` can lead to a value having 2 two owners -&gt; double free
+-   `unsafe` can make immutable data temporarily mutable, which will lead to broken promises and tears.
+
+---
+
+Rust allows you to shoot yourself in the foot, it just requires you to get the gun out of the holster and remove the safety before.
```
Comment (Hoverbear) [inline presentation/chapters/en-US/unsafe.chapter]: "to take your gun out of the holster and remove the safety first"

File: presentation/chapters/en-US/ownership.chapter [hunk 31]
```diff
@@ -1 +1,153 @@
...
+
+<pre><code data-source="chapters/shared/code/ownership/2.rs" data-trim="hljs rust"></code></pre>
+
+---
+
+## Oops!
+
+---
+
+In Rust-Lingo, this is call `consuming`. `pacman` cosumes `dot`.
```
Comment (ericho) [inline presentation/chapters/en-US/ownership.chapter]: Typo on "cosumes"

**Pull request merged**